### PR TITLE
[Core] Remove `ManagerInterface.thumbnailSpecification`

### PR DESCRIPTION
**Pull request by foundrytom**
Implements the approach proposed in #86. It's just plain simple better. Thumbnails no longer need bespoke API, and now just use the standard query/publish semantics applicable to any other entity.

This updates the documentation and adds an example of the correct implementation in a host.

### CODE REVIEW

File: doc/src/Thumbnails.dox [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ * to look up a thumbnail for an existing entity.
+ *
+ * @section thumbnails_creation_during_publish Requesting Thumbnail Creation During Publishing
+ *
+ * In many situations, it is difficult for a @ref manager to
+ * retrospectively generate a thumbnail for any given @ref entity. It
+ * usually requires first-class knowledge of how the specific entity
+ * type can be rendered to into an image.
+ *
+ * For image-based entities, this is potentially straight forward. The
```
Comment (feltech) [inline doc/src/Thumbnails.dox:17]: [fix]
```suggestion
 * For image-based entities, this is potentially straightforward. The
```

File: doc/src/Thumbnails.dox [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ * - Call @ref openassetio.managerAPI.ManagerInterface.ManagerInterface.getEntityMetadata
+ *   "getEntityMetadata" on the returned reference to determine the desired width/height/format
+ *   for the thumbnail.
+ * - Generate a thumbnail and @ref register it to the same reference.
+ *
+ * @note The thumbnail generation process may be carried out
+ * asynchronously by the host and registration may occur at an arbitrary
+ * time in the future.
+ *
+ * @see example_generating_a_thumbnail
```
Comment (feltech) [inline doc/src/Thumbnails.dox:42]: [fix] This ref doesn't render.

File: python/openassetio/constants.py [hunk 3]
```diff
@@ -48,6 +48,10 @@
 ## will determine the URL to use for writing data for new entities
 kWillManagePath = 8
 
+## If True, and supported by a @ref host, a thumbnail will be generated by the
```
Comment (feltech) [inline python/openassetio/constants.py:51]: [fix] 
```suggestion
## If True, and supported by a @ref host, a thumbnail will be generated by the
```
Subtle NBSP (how did you manage that?), but does affect the rendered output.

File: doc/src/Thumbnails.dox [hunk 4]
```diff
@@ -0,0 +1,57 @@
+/*!
+ * @page thumbnails Thumbnails
+ *
+ * @section thumbnails_overview Overview
+ *
+ * OAIO provides a mechanism for a @ref manager to request a thumbnail
+ * from the @ref host when a new @ref entity is created, and for a host
+ * to look up a thumbnail for an existing entity.
+ *
+ * @section thumbnails_creation_during_publish Requesting Thumbnail Creation During Publishing
```
Comment (feltech) [inline doc/src/Thumbnails.dox:10]: [consider][minor] This title gets cut off (well, causes a horizontal scrollbar) in the sidebar. Very minor quibble.

**Review by feltech — CHANGES_REQUESTED** [review]

File: doc/src/Thumbnails.dox [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+ *
+ * @note The thumbnail generation process may be carried out
+ * asynchronously by the host and registration may occur at an arbitrary
+ * time in the future.
+ *
+ * @see example_generating_a_thumbnail
+ *
+ * @todo Document the attributes used to determine width/hight/format,
+ * these should be the same as the fields of the
+ * ThumbnailSpecification.
```
Comment (feltech) [inline doc/src/Thumbnails.dox:46]: [question] What is the interaction between width/height/format in the Specification vs. the Attributes?  I can understand the attributes are what the manager wants. So what's the use case for having these fields also in the specification?  Is the idea perhaps that a host can specify a list of formats and resolutions it supports, then the attributes given by the manager are selected from that list?

Comment (foundrytom) [reply doc/src/Thumbnails.dox:46]: Yeah that thing, and when (in the future) a host requests a thumbnail from a manager via `getRelatedReferences`, they can request a specific format/size.

File: doc/src/Examples.dox [hunk 6]
```diff
@@ -202,4 +202,48 @@
  * with open(os.path.join(os.path.expanduser('~'), 'history', 'a') as f:
  *     f.write(f"{final_ref}\n")
  * @endcode
+ *
+ * @section example_generating_a_thumbnail Generating a Thumbnail During Publish
```
[the remark this answers is not in the document]
Comment (foundrytom) [reply doc/src/Examples.dox:206]: Saving this for later, as we have other long titles.

File: doc/src/Thumbnails.dox (hunk 4 above)
Comment (foundrytom) [reply doc/src/Thumbnails.dox:10]: Saving this for later, as we have other long titles.

File: doc/src/Thumbnails.dox (hunk 2 above)
Comment (foundrytom) [reply doc/src/Thumbnails.dox]: Fixed [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/116/commits/f16b1413aaa134ce202f67352f1fa6e597ebf15a).

File: python/openassetio/constants.py (hunk 3 above)
Comment (foundrytom) [reply python/openassetio/constants.py]: Fixed [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/116/commits/34e2fdf021f156a3c1c9328c61b4027d3b1378b1)

**Review by feltech — APPROVED** [review]
LGTM squash and merge when ready

**Pull request merged**